### ISSUE TITLE
Sporadic failure initializing helm client with gke clusters

### ISSUE THREAD

**Issue by mjhuber** [FairwindsOps/reckoner#147]
Using reckoner 2.0.0 I sometimes receive an error from reckoner initializing the helm client when using with gke clusters.  To workaround this I am able to run an arbitrary helm command after receiving the error, and then subsequent `reckoner` commands succeed.  Steps to reproduce:

1.  Run a reckoner plot command (ex. `reckoner plot course.yml --heading nginx-ingress`).
2.  Reckoner outputs an error:
```
⛵🔥 Encountered errors while reading course file ⛵🔥
Helm Client Failed to initialize: Command Failed with output below:
STDOUT:
STDERR: Error: Get https://35.197.78.144/api/v1/namespaces/helm-system/pods?labelSelector=app%3Dhelm%2Cname%3Dtiller: error executing access token command "/Users/huberm/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output= stderr=Traceback (most recent call last):
  File "/Users/huberm/google-cloud-sdk/lib/gcloud.py", line 20, in <module>
    from __future__ import absolute_import
ImportError: No module named __future__


COMMAND: helm version --short --server
```
3.  Run an arbitrary helm command (e.g. `helm version`).
4.  Run `reckoner plot` command again, no failure.

```
$ reckoner version
2.0.0
```

```
$helm version
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```

**Comment by bambash** [comment]
We ran into this while testing helm3 changes. I believe the problem is with the way gcloud handles cluster token authentication... it's been a little while and my memory is fuzzy. I think it also has to do with subshells? @endzyme do you recall the exact issue?

**Comment by endzyme** [comment]
Found the root cause - @ivanfetch is working on an internal fix but here's the long and short so far:

Reckoner uses PyInstaller to build binaries made from python files. Our internal tooling uses `virtual_env` python binaries to try to keep environments and laptops clean. When Reckoner runs, it automatically removes the `VIRTUAL_ENV` environment variable to try and protect itself from module leaking (presumably -- see [this line in the source](https://github.com/pyinstaller/pyinstaller/blob/bdbf99d7327b13741d63eba70c4f0031e4997a14/PyInstaller/loader/pyiboot01_bootstrap.py#L47-L55)). When PyInstaller/Reckoner removes the `VIRTUAL_ENV` variable, it still defaults to finding `python2` binary in the `venv` path.

When `gcloud` looks for `which python2`, it returns `/path/of/your/.venv/bin/python2`. Also since `VIRTUAL_ENV` env var is NOT set, the `gcloud` command thinks you're using a fully built distribution of python with all system site modules available it. Unfortunately this is not the case since what is actually happening is Reckoner (via PyInstaller) is unsetting the env var but still using the virtual env binary (which doesn't have access to system site modules). This causes `python2 -S -c 'import __future__'` to fail when `VIRTUAL_ENV` is _not_ set but when still using the virutal env's build of python2. 

Long story short - to "fix" this you can set `CLOUDSDK_PYTHON=/usr/bin/python` (or your location) and it avoids using the broken python version in a virtual env.

There is also an [open PR](https://github.com/pyinstaller/pyinstaller/pull/4401) to add flags to NOT delete this env var.

**Comment by ejether** [comment]
This is evil. Great work tracking it down!
